financial-connections/src/main/java/com/stripe/android/financialconnections/browser/BrowserManager.kt

```diff
@@ -0,0 +1,58 @@
+package com.stripe.android.financialconnections.browser
+
+import android.app.Application
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import androidx.browser.customtabs.CustomTabsIntent
+import javax.inject.Inject
+
+internal class BrowserManager @Inject constructor(
+    val context: Application
+) {
+
+    /**
+     * Returns `true` if there's an app that can open an http url.
+     */
+    fun canOpenHttpUrl(): Boolean = getPackageToHandleUri(Uri.parse("https://")) == null
+
+    /**
+     * Get the default browser app package that will be used to open the given [Uri].
+     */
+    fun getPackageToHandleUri(uri: Uri): String? = runCatching {
+        getPackageToHandleIntent(uri.toIntent())
+    }.getOrNull()
+
+    /**
+     * Constructs an intent to launch the given [Uri] on a CustomTab or in a regular browser,
+     * based on the default browser set for this device.
+     */
+    fun createBrowserIntentForUrl(uri: Uri): Intent {
+        val browserIntent = uri.toIntent()
+        val defaultPackage = getPackageToHandleIntent(browserIntent)
+        return when {
+            /**
+             * Firefox browser has a redirect issue when launching as a custom tab.
+             * @see [BANKCON-3846]
+             */
+            defaultPackage?.contains(FIREFOX_PACKAGE) == true -> browserIntent
+            else -> createCustomTabIntent(uri)
```

Comment:
What will happen if no Chrome and Firefox are installed? Do we need to add another fallback behavior?

Reply:
custom tabs are not chrome exclusive - some browsers support it and some others dont. (see [this](https://developer.chrome.com/docs/android/custom-tabs/browser-support/))

Firefox is one of these browsers with custom tabs support, but we've an issue with one of our aggregators where redirects don't work on their custom tabs implementation.

For chrome (and all the other browsers) we build a custom tabs intent, that falls back to open on the browser if the default browser does not support custom tabs.